crates/turborepo-lib/src/task_graph/visitor.rs

```diff
@@ -114,7 +115,11 @@ impl<'a> Visitor<'a> {
         let mut tasks = FuturesUnordered::new();
         let errors = Arc::new(Mutex::new(Vec::new()));
 
+        let span = Span::current();
+
         while let Some(message) = node_stream.recv().await {
+            let span = tracing::debug_span!(parent: &span, "queue_task", task = %message.info);
+            let _enter = span.enter();
```

Comment:
Do we want to drop this here so we can measure how long we block the main thread before moving the work to a new task?

Reply:
The amount of work done after spawning the future is so low that we can probably get away with not creating an additional span for it.